optin/reporting/test/utils/Validation.spec.js

```diff
@@ -0,0 +1,373 @@
+const Validation = require('../../utils/Validation');
+const TestUtils = require('../TestUtils');
+const { expect } = require('chai');
+const { validationResult } = require('express-validator');
+
+const mockExpressValidatorMiddleware = async (req, res, middlewares) => {
+	await Promise.all(middlewares.map(async middleware => {
+		await middleware(req, res, () => undefined);
+	}));
+};
+
+const runValidAsserts = async ({ validator, validParams, queryField }) => {
+	/* eslint-disable no-await-in-loop */
+	for (let i = 0; i <= validParams.length - 1; ++i) {
+		// Arrange:
+		const req = TestUtils.mockRequest({
+			pageSize: 10,
+			pageNumber: 1,
+			[queryField]: validParams[i]
+		});
+		const res = TestUtils.mockResponse();
+
+		// Act:
+		await mockExpressValidatorMiddleware(req, res, validator);
+		const { errors } = validationResult(req);
+
+		// Assert:
+		expect(errors.length).to.equal(0);
+	}
+};
+
+const runInvalidAsserts = async ({
+	validator, invalidParams, queryField, errorMessage
+}) => {
+	/* eslint-disable no-await-in-loop */
+	for (let i = 0; i <= invalidParams.length - 1; ++i) {
+		// Arrange:
+		const req = TestUtils.mockRequest({
+			pageSize: 10,
+			pageNumber: 1,
+			[queryField]: invalidParams[i]
+		});
+
+		const res = TestUtils.mockResponse();
+
+		// Act:
+		await mockExpressValidatorMiddleware(req, res, validator);
+		const { errors } = validationResult(req);
+
+		// Assert:
+		expect(errors[0].msg).to.equal(errorMessage);
+	}
+};
```

Comment:
only difference between these two functions is expect param, so consider refactoring